### PR TITLE
Feature/MOB-3957 quote list sort view

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
## JIRA
[MOB-3957](https://karhoo.atlassian.net/browse/MOB-3957)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/Screens/QuoteList/QuoteListCoordinator.swift [hunk 1]
```diff
@@ -56,35 +56,20 @@ final class KarhooQuoteListCoordinator: QuoteListCoordinator {
 extension KarhooQuoteListCoordinator: QuoteListRouter {
 
     func routeToQuote(_ quote: Quote, journeyDetails: JourneyDetails) {
-        // TODO: replace to use Booking coordinator to perform actual navigation
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Screens/QuoteList/QuoteListCoordinator.swift:59]: Do we still need this block of commented  code?

File: KarhooUISDK/Screens/QuoteList/QuoteListViewController.swift [hunk 2]
```diff
@@ -30,24 +30,49 @@ final class KarhooQuoteListViewController: UIViewController, BaseViewController,
...
-        $0.alignment = .fill
+        $0.alignment = .center
         $0.spacing = UIConstants.Spacing.standard
     }
     private var headerViews: [UIView] {
-        [
+        #if DEBUG
+        return [
+            addressPickerView,
+            buttonsStackView,
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Screens/QuoteList/QuoteListViewController.swift:40]: why we need to return different objects for debug/production? Next uisdk version will be released when sorting and filtering will be implemented.

File: KarhooUISDK.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -366,6 +366,11 @@
...
 		511D9CCE27D90193004CBA6B /* UINAvigationController+navigationCompletions.swift in Sources */ = {isa = PBXBuildFile; fileRef = 511D9CCD27D90193004CBA6B /* UINAvigationController+navigationCompletions.swift */; };
+		5132EAF627EBA6F100866FA9 /* MainActionButton.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5132EAF527EBA6F000866FA9 /* MainActionButton.swift */; };
+		5132EAF927ECD02500866FA9 /* SingleSelectionListView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5132EAF827ECD02500866FA9 /* SingleSelectionListView.swift */; };
+		5132EAFC27ECF5E100866FA9 /* UserSelectable.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5132EAFB27ECF5E100866FA9 /* UserSelectable.swift */; };
+		5132EB0227F1E1D000866FA9 /* RadioControl.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5132EB0127F1E1D000866FA9 /* RadioControl.swift */; };
+		5132EB0427F2517B00866FA9 /* UIControl+addTouchAnimation.swift in Sources */ = {isa = PBXBuildFile; fileRef = 5132EB0327F2517B00866FA9 /* UIControl+addTouchAnimation.swift */; };
```
Comment (dianakarh) [inline KarhooUISDK.xcodeproj/project.pbxproj:373]: I think `addTouchAnimation` was meant to be `AddTouchAnimation` with a capital letter, right?

File: KarhooUISDK/CommonUI/Views/RadioControl/RadioControl.swift [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        set {
+            setSelected(newValue)
+        }
+    }
+
+    /// Set false if view should not allow touch interactions to change it's state to unselected
+    let unselectAllowed: Bool
+
+    /// Local value to store current state, since `isSelected` is used by UIKit and need to be computed property.
+    private var selectionStatus: Bool = false
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/Views/RadioControl/RadioControl.swift:44]: `selectionStatus` sounds more like an enum than a bool. What about `isOn`?

Comment (dianakarh) [inline KarhooUISDK/CommonUI/Views/RadioControl/RadioControl.swift:44]: `selectionStatus` sounds more like an enum than a bool. What about `isOn`?

File: KarhooUISDK/CommonUI/Views/SingleSelectionListView/SingleSelectionListView.swift [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    }
+}
+
+// MARK: - SelectionRowView
+private class SelectionRowView<T: UserSelectable>: UIView {
+    let value: T
+    let onSelected: (T) -> Void
+
+    /// Set `true` if it is allowed to have non of the options selected.
+    let allowedToUnselectItself: Bool
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/Views/SingleSelectionListView/SingleSelectionListView.swift:103]: Why not use the same name for it, i.e.: `unselectAllowed`?

**Review by BartlomiejSopalaKarhoo — APPROVED** [review]

File: KarhooUISDK/Screens/QuoteList/QuoteListViewController.swift [hunk 6]
```diff
@@ -132,6 +148,7 @@ final class KarhooQuoteListViewController: UIViewController, BaseViewController,
         let tableViewController = tableViewCoordinator.viewController
         view.addSubview(tableViewController.view)
         addChild(tableViewController)
+        buttonsStackView.addArrangedSubviews([sortButton])
```
Comment (BartlomiejSopalaKarhoo) [inline KarhooUISDK/Screens/QuoteList/QuoteListViewController.swift:151]: sort button should by hidden when user is planning trip in future

**Review by BartlomiejSopalaKarhoo — CHANGES_REQUESTED** [review]

**Review by BartlomiejSopalaKarhoo — APPROVED** [review]

**Pull request merged**